apm-sniffer/optional-plugins/optional-spring-plugins/spring-webflux-5.x-reactive-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/reactive/DispatcherHandlerHandleMethodInterceptor.java

```diff
@@ -16,62 +16,90 @@
+    private ContextManagerExtendService extendService = ServiceManager.INSTANCE.findService(ContextManagerExtendService.class);
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                              MethodInterceptResult result) throws Throwable {
-        EnhancedInstance instance = DispatcherHandlerHandleMethodInterceptor.getInstance(allArguments[0]);
-        if (instance != null) {
-            ContextCarrier contextCarrier = new ContextCarrier();
-            CarrierItem next = contextCarrier.items();
-            ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
-            HttpHeaders headers = exchange.getRequest().getHeaders();
-            while (next.hasNext()) {
-                next = next.next();
-                List<String> header = headers.get(next.getHeadKey());
-                if (header != null && header.size() > 0) {
-                    next.setHeadValue(header.get(0));
-                }
-            }
-            AbstractSpan span = ContextManager.createEntrySpan(WIP_OPERATION_NAME, contextCarrier);
-            span.setComponent(ComponentsDefine.SPRING_WEBFLUX);
-            SpanLayer.asHttp(span);
-            Tags.URL.set(span, exchange.getRequest().getURI().toString());
-            instance.setSkyWalkingDynamicField(span);
-        }
+
     }
 
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                               Object ret) throws Throwable {
-        return ret;
+        EnhancedInstance instance = getInstance(allArguments[0]);
+
+        ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
+
+        ContextCarrier carrier = new ContextCarrier();
+        CarrierItem next = carrier.items();
+        HttpHeaders headers = exchange.getRequest().getHeaders();
+        while (next.hasNext()) {
+            next = next.next();
+            List<String> header = headers.get(next.getHeadKey());
+            if (header != null && header.size() > 0) {
+                next.setHeadValue(header.get(0));
+            }
+        }
+
+        String pathStr = exchange.getRequest().getPath().value();
+        String methodStr = exchange.getRequest().getMethodValue();
+        String operationName = methodStr + ":" + pathStr;
+
+        AbstractTracerContext context = extendService.createTraceContext(operationName, false);
+        AbstractSpan span = context.createEntrySpan(operationName);
+        if (carrier != null && carrier.isValid()) {
+            context.extract(carrier);
+        }
+
+        span.setComponent(ComponentsDefine.SPRING_WEBFLUX);
+        SpanLayer.asHttp(span);
+        Tags.URL.set(span, exchange.getRequest().getURI().toString());
+        HTTP.METHOD.set(span,methodStr);
+
+        instance.setSkyWalkingDynamicField(context);
```

Comment:
Why set context , does it useful?   
Currently `gateway plugin` rely on value `span` of `SkyWalkingDynamicField` . 
if it's useful ,maybe need to update `gateway plugin`

Reply:
because ``Context`` cannot be passed through ``ThreadLocal``. Downstream needs to get ``Context`` from ``SkyWalkingDynamic`` .   Reactive programming, the thread of processing is changing

Reply:
> because `Context` cannot be passed through `ThreadLocal`. Downstream needs to get `Context` from `SkyWalkingDynamic` . Reactive programming, the thread of processing is changing
1. yes, it cannot pass, But what do Downstream use to 'Context'.  Maybe  `ContextSnapshot`  is more useful when cross thread 
2. if u do. u should modify `FilteringWebHandlerInterceptor.java`，because it need `span` ,not `Context` Currently。

Reply:
I see it. gateway relies on webflux. Let me try